engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java

```diff
@@ -1871,6 +1908,153 @@ public static long minus(@Nullable final ZonedDateTime dateTime1, @Nullable fina
         return checkUnderflowMinus(epochNanos(dateTime1), epochNanos(dateTime2), true);
     }
 
+    /**
+     * Subtracts two durations.
+     *
+     * @param duration1 first duration
+     * @param duration2 second duration
+     * @return {@code null} if either input is {@code null}; otherwise the difference of the two durations
+     */
+    public static Duration minus(@Nullable final Duration duration1, @Nullable final Duration duration2) {
+        if (duration1 == null || duration2 == null) {
+            return null;
+        }
+
+        try {
+            return duration1.minus(duration2);
+        } catch (Exception ex) {
+            throw new DateTimeOverflowException(ex);
+        }
+    }
+
+    /**
+     * Subtracts two periods.
+     *
+     * @param period1 first period
+     * @param period2 second period
+     * @return {@code null} if either input is {@code null}; otherwise the difference of the two periods
+     */
+    public static Period minus(@Nullable final Period period1, @Nullable final Period period2) {
+        if (period1 == null || period2 == null) {
+            return null;
+        }
+
+        try {
+            return period1.minus(period2);
+        } catch (Exception ex) {
+            throw new DateTimeOverflowException(ex);
+        }
+    }
+
+    /**
+     * Multiply a duration by a scalar.
+     *
+     * @param duration the duration to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the duration multiplied by the scalar
+     */
+    public static Duration multiply(final Duration duration, final long scalar) {
+        if (duration == null || scalar == NULL_LONG) {
+            return null;
+        }
+
+        try {
+            return duration.multipliedBy(scalar);
+        } catch (ArithmeticException ex) {
+            throw new DateTimeOverflowException(ex);
+        }
```

Comment:
This is inconsistent (although probably more correct) than the versions that `catch (Exception ex)`. The date time operations typically document throwing:

```java
     * @throws DateTimeException if the subtraction cannot be made
     * @throws ArithmeticException if numeric overflow occurs
```

So, when we catch `Exception` and re-wrap in `DateTimeOverflowException`, we might be "lying" b/c this could be caused by something other than overflow.

I see this pattern is used widely in DateTimeUtils, I think imprecisely.

Reply:
I have adjusted the handling to be:
```java
        } catch (DateTimeException | ArithmeticException ex) {
            throw new DateTimeOverflowException(ex);
        }
```
Overflows can come through either exception.  `java.time` does use `DateTimeException` for other problems, but those problems should not be in the methods here.